integration-tests/test_status.py

```diff
@@ -0,0 +1,42 @@
+import contextlib
+import pytest
+from time import sleep
+
+pytestmark = pytest.mark.usefixtures("register_subman")
+
+
+def test_status_registered(external_candlepin, insights_client):
+    """
+    Test `insights-client --status` when the client is registered;
+    in case of legacy_upload status command returns a different output.
+
+    """
+    insights_client.register()
+    assert insights_client.is_registered
+    # Adding a small wait to ensure inventory is up-to-date
+    sleep(5)
+    registration_status = insights_client.run("--status")
+    if insights_client.config.legacy_upload:
+        assert registration_status.returncode == 0
```

Comment:
this check is not needed: `InsightsClient.run()` by default has `check=True`, which means it will fail when the return code of `insights-client` is different than 0